spec/unit/outputs/elasticsearch/http_client_spec.rb

```diff
@@ -266,6 +266,18 @@
             end
           end
         end
+
+        context "with partial unsupported action" do
+          let(:actions) { [
+            ["index", {:_id=>nil, :_index=>"logstash"}, {"message"=> "hello"}],
+            ["unsupported_action", {:_id=>nil, :_index=>"logstash"}, {"message"=> "world!"}],
+          ]}
+          it "executes one bulk_send operation" do
+            allow(subject).to receive(:join_bulk_responses)
+            expect(subject).to receive(:bulk_send).once
+            s = subject.send(:bulk, actions)
+          end
+        end
```

Comment:
we should also make assertions about what we expect back. The emulation of an error response and zipping back together with the actual bulk requests that are sent are currently untested.

Reply:
please take a look at changes.